src/components/assessment/AudioAssessment.jsx

```
@@ -30,41 +21,47 @@ const AudioAssessment = ({
+    },
+    []
+  );
 
-  // Apply audio quality check to each audio source
-  audioSources?.forEach(audioUrl => {
-    useAudioQuality(audioUrl, handleQualityCheck(audioUrl));
-  });
+  // ✅ Run audio quality hook inside an effect, not inside forEach with hooks
+  useEffect(() => {
+    audioSources.forEach((audioUrl) => {
+      useAudioQuality(audioUrl, handleQualityCheck(audioUrl));
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)